Update journal_name.py

HesongHuang opened the pull request:
put the data into a diction,like {"@article1": "hardin2014clinical,", "  title=1": "The clinical and genetic features of COPD-asthma overlap syndrome},", "  author=1": "Hardin, Megan and Cho, Michael and McDonald, Merry-Lynn and Beaty, Terri and Ramsdell, Joe and Bhatt, Surya and Van Beek, Edwin JR and Make, Barry J and Crapo, James D and Silverman, Edwin K and others},", "  journal=1": "European Respiratory Journal},", "  volume=1": "44},", "  number=1": "2},", "  pages=1": "341--350},", "  year=1": "2014},", "  publisher=1": "Eur Respiratory Soc}", "@article2": "lasso2014plus,",} the number "=1"represent they are from the same article.
Save the diction into a new json file called "Bibtext_format.json"
I have to say, when you input -c 100, you actually get 200 results,besauce every page of google scholar shows 20 results.
So the number of journal is true.

HesongHuang committed:
Update journal_name.py

put the data into a diction,like {"@article1": "hardin2014clinical,", "  title=1": "The clinical and genetic features of COPD-asthma overlap syndrome},", "  author=1": "Hardin, Megan and Cho, Michael and McDonald, Merry-Lynn and Beaty, Terri and Ramsdell, Joe and Bhatt, Surya and Van Beek, Edwin JR and Make, Barry J and Crapo, James D and Silverman, Edwin K and others},", "  journal=1": "European Respiratory Journal},", "  volume=1": "44},", "  number=1": "2},", "  pages=1": "341--350},", "  year=1": "2014},", "  publisher=1": "Eur Respiratory Soc}", "@article2": "lasso2014plus,",} the number "=1"represent they are from the same article.
Save the diction into a new json file called "Bibtext_format.json"
I have to say, when you input -c 100, you actually get 200 results,besauce every page of google scholar shows 20 results.
So the number of journal is true.

fedorov commented:
This PR aims to resolve issue #5

fedorov commented on journal_name.py:
```diff
@@ -3,33 +3,50 @@
+        json_file.write(json.dumps(data))
+
+        
 def get_json():
```
This is not a good approach to implement this functionality. In this same program, you save the JSON file, and then read it again, and then save another JSON.

Please change your implementation approach to have `get_json` function take input parameter with the input as string, and return json dictionary. Don't save json string and then read and save dictionary json. Pass the bibtex string you get from Google Scholar to the function, and save just the result dictionary.

fedorov commented on journal_name.py:
```diff
@@ -3,33 +3,50 @@
-			b = a.split('\n')
-			for m in b: 
-				if "journal={" in m:
+			c=a.split("\n")  #type of data is list
```
Don't split by "\n". Your input looks like this:

```
@article{hardin2014clinical, title={The clinical and genetic features of COPD-asthma overlap syndrome}, 
author={Hardin, Megan and Cho, Michael and McDonald, Merry-Lynn and Beaty, Terri and Ramsdell, 
Joe and Bhatt, Surya and Van Beek, Edwin JR and Make, Barry J and Crapo, James D and Silverman, 
Edwin K and others}, journal={European Respiratory Journal}, volume={44}, number={2}, pages={341--350}, 
year={2014}, publisher={Eur Respiratory Soc} } "
```

Text between `@` and `{` is the article type. Next take everything inside `{ }` and split by comma, remove the first item from the result, then split each item by `=`, the first item in the result will be the attribute name, the second item, after removing `{ }` will be the attribute value.

HesongHuang committed:
#5 Updata

store the data into a txt, and type of the data is string, looks like:
@article{hardin2014clinical,
  title={The clinical and genetic features of COPD-asthma overlap syndrome},
  author={Hardin, Megan and Cho, Michael and McDonald, Merry-Lynn and Beaty, Terri and Ramsdell, Joe and Bhatt, Surya and Van Beek, Edwin JR and Make, Barry J and Crapo, James D and Silverman, Edwin K and others},
  journal={European Respiratory Journal},
  volume={44},
  number={2},
  pages={341--350},
  year={2014},
  publisher={Eur Respiratory Soc}
}

HesongHuang committed:
Update journal_name.py

Make a better json file, like:
{"  title": "{Validation of Model-Based Brain Shift Correction in Neurosurgery via Intraoperative Magnetic Resonance Imaging: Preliminary Results}", "  author": "{Luo, Ma and Frisken, Sarah F and Weis, Jared A and Clements, Logan W and Unadkat, Prashin and Thompson, Reid C and Golby, Alexandra J and Miga, Michael I}", "  journal": "{PloS one}", "  volume": "{12}", "  number": "{1}", "  pages": "{1013503--1013503}", "  year": "{2017}", "  publisher": "{Public Library of Science}\n}\n", "  booktitle": "{SPIE Medical Imaging}", "  organization": "{International Society for Optics and Photonics}\n}\n", "  institution": "{Tech. Rep.(Technische Universit{\\\"a}t M{\\\"u}nchen, 2014)}\n}\n", "  school": "{The University of Western Ontario}\n}\n"}
Delete the poor demo of the results and plan to use html to make a representation.

HesongHuang committed:
Update bibtex.json

New json file

HesongHuang committed:
Update bibtex.json

HesongHuang committed:
Update journal_name.py

Get the title of every article in a desperate json file.
